inference-engine/include/ie_parameter.hpp

```diff
@@ -83,7 +83,8 @@ class INFERENCE_ENGINE_API_CLASS(Parameter) {
      * @param parameter object
      */
     template <class T,
-              typename = typename std::enable_if<!std::is_same<typename std::decay<T>::type, Parameter>::value>::type>
+              typename = typename std::enable_if<!std::is_same<typename std::decay<T>::type, Parameter>::value &&
+                                                 !std::is_abstract<typename std::decay<T>::type>::value>::type>
```

Comment:
@ilyachur please take a look at this particular change. Explanation you can find in corresponding commit [commit](https://github.com/openvinotoolkit/openvino/pull/3211/commits/d28b80136a202d942a63b327333838177a7c1c27)

Reply:
Thank you for explanation! This change LGTM